src/main/java/jenkins/scm/api/SCMFile.java

```diff
@@ -40,6 +40,16 @@
  * @author Kohsuke Kawaguchi
  */
 public abstract class SCMFile {
+
+    /**
+     * Cache of the last modified timestamp, to allow repeated calls to minimize the number of network round trips.
+     */
+    private Long modified;
```

Comment:
Should any caching not be the responsibility of the implementation? After all, it may have a better way of doing so. `GitSCMSource` in particular can do metadata lookups cheaply enough that there is no point in caching them. And even a remote-based source may have a better way of keeping a cache.